diskquota_utility.c

```diff
@@ -1163,29 +1164,25 @@ get_rel_oid_list(void)
 		HeapTuple	tup;
 		bool        	isnull;
 		Oid         	oid;
-		ListCell   	*l;
+		char 			relkind;
 
 		tup = SPI_tuptable->vals[i];
 		oid = DatumGetObjectId(SPI_getbinval(tup,tupdesc, 1, &isnull));
+		relkind = DatumGetChar(SPI_getbinval(tup,tupdesc, 2, &isnull));
 		if (!isnull)
 		{
-			Relation	relation;
-			List	   	*indexIds;
-			relation = try_relation_open(oid, AccessShareLock, false);
-			if (!relation)
-				continue;
-
-			oidlist = lappend_oid(oidlist, oid);
-			indexIds = RelationGetIndexList(relation);
-			if (indexIds != NIL )
+			if (relkind == RELKIND_INDEX)
 			{
-				foreach(l, indexIds)
-				{
-					oidlist = lappend_oid(oidlist, lfirst_oid(l));
+				HeapTuple index_tup = SearchSysCacheCopy1(INDEXRELID, ObjectIdGetDatum(oid));
+				if (HeapTupleIsValid(index_tup)) {
+					Oid table_oid = DatumGetObjectId(SysCacheGetAttr(INDEXRELID, index_tup, Anum_pg_index_indrelid, &isnull));
+					HeapTuple table_tup = SearchSysCacheCopy1(RELOID, table_oid);
+					char table_kind = ((Form_pg_class) GETSTRUCT(table_tup))->relkind;
+					if (table_kind != RELKIND_RELATION || table_kind != RELKIND_MATVIEW)
```

Comment:
I think it should be `&&`.

```suggestion
					if (table_kind != RELKIND_RELATION && table_kind != RELKIND_MATVIEW)
```